src/debug/foundry_cheatcodes.ts

```diff
@@ -46,20 +102,161 @@ export const FAIL_MSG_DATA = Buffer.concat([
     setLengthLeft(Buffer.from([1]), 32)
 ]).toString("hex");
 
+export interface FoundryPrank {
+    sender: Address;
+    origin: Address | undefined;
+    once: boolean;
+}
+
+/**
+ * Helper class attached to each VM object, that contains state relevant to Foundry cheatcodes.
+ */
 export class FoundryContext {
     public timeWarp: bigint | undefined = undefined;
     public rollBockNum: bigint | undefined = undefined;
     public failCalled = false;
+
+    /**
+     * We keep track of the current stack of EEI objects, in order to keep track
+     * of any pranks (pending and current).  Its easier to attach pranks to EEI
+     * objects, as there is a unique EEI object for each external call in the
+     * trace, and pranks are scoped to external calls.
+     */
+    private eeiStack: EEI[] = [];
+
+    // Get the current (topmost) EEI object
+    public getEEI(): EEI {
+        return this.eeiStack[this.eeiStack.length - 1];
+    }
+
+    // Get any pending pranks for the current call frame
+    public getPendingPrank(): FoundryPrank | undefined {
+        if (this.eeiStack.length === 0) {
+            return undefined;
+        }
+
+        const eei = this.eeiStack[this.eeiStack.length - 1];
+
+        return (eei as any).pendingPrank;
+    }
+
+    // Set the pending prank for the current call frame
+    public setPendingPrank(prank: FoundryPrank | undefined): void {
+        if (this.eeiStack.length === 0) {
+            return undefined;
+        }
+
+        const eei = this.eeiStack[this.eeiStack.length - 1];
+
+        (eei as any).pendingPrank = prank;
```

Comment:
This also can be shortened with `this.getEEI()`.

Reply:
Fixed in [b29dbfe](https://github.com/ConsenSys/sol-dbg/pull/23/commits/b29dbfea1af90e8af6a85bc2fd2f4e67e6a8ca8a)